### PR TITLE
fix(anvil): handle zero prune history safely

### PR DESCRIPTION
**Pull request by mk0walsk**
## Motivation

`--prune-history 0` currently reaches the in-memory history cache as `0`. `InMemoryBlockStates::enforce_limits()` then loops on `while self.present.len() >= self.in_memory_limit`, which can hang block production when the limit is zero.

## Solution

Normalize `max_memory_history == 0` to the unsupported path in `PruneStateHistoryConfig`, and defensively clamp in-memory cache limits to at least `1` inside `InMemoryBlockStates`. Add regression tests for the config normalization and zero-limit cache guard.

## PR Checklist

- [x] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/anvil/src/eth/backend/mem/storage.rs [hunk 1]
```diff
@@ -198,7 +199,10 @@ impl InMemoryBlockStates {
 
     /// Sets the maximum number of stats we keep in memory
     pub const fn set_cache_limit(&mut self, limit: usize) {
+        let limit = if limit == 0 { 1 } else { limit };
```
Comment (mablr) [inline crates/anvil/src/eth/backend/mem/storage.rs:202]:
```suggestion
        let limit = limit.max(1);
```
like above

**Review by mablr — COMMENTED** [review]
sgtm

**Comment by mk0walsk** [comment]
@mablr Done, please take another look, thanks!

File: crates/anvil/src/eth/backend/mem/storage.rs [hunk 2]
```diff
@@ -70,8 +70,13 @@ pub struct InMemoryBlockStates {
 }
 
 impl InMemoryBlockStates {
+    const fn normalize_in_memory_limit(limit: usize) -> usize {
+        if limit == 0 { 1 } else { limit }
+    }
```
Comment (mablr) [inline crates/anvil/src/eth/backend/mem/storage.rs]: this can be free function, or inline btw

**Review by mablr — COMMENTED** [review]
smol pedantic nit :)

otherwise it's good.

File: crates/anvil/src/eth/backend/mem/storage.rs (hunk 2 above)
Comment (mablr) [reply crates/anvil/src/eth/backend/mem/storage.rs]: initial `.max(1)` was good 👍

Comment (mk0walsk) [reply crates/anvil/src/eth/backend/mem/storage.rs]: Done, please re-check. The helper is gone and the clamp is now inlined.

**Review by mablr — DISMISSED** [review]
sgtm, pending another review ...

File: crates/anvil/src/eth/backend/mem/storage.rs [hunk 3]
```diff
@@ -203,7 +199,7 @@ impl InMemoryBlockStates {
 
     /// Sets the maximum number of stats we keep in memory
     pub const fn set_cache_limit(&mut self, limit: usize) {
-        let limit = Self::normalize_in_memory_limit(limit);
+        let limit = if limit == 0 { 1 } else { limit };
```
Comment (figtracer) [inline crates/anvil/src/eth/backend/mem/storage.rs:202]: can't we use max(1) here too?

Comment (mk0walsk) [reply crates/anvil/src/eth/backend/mem/storage.rs:202]: Sure, updated to use `max(1)` there too. Let me know if anything still looks off.Sure, updated to use `max(1)` there too. Let me know if anything still looks off.

**Review by figtracer — APPROVED** [review]

**Review by mablr — APPROVED** [review]
